Downloader.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  downloadModel.swift
+//  react-native-spokestack
+//
+//  Created by Timothy Willison on 12/7/20.
+//
+
+import Foundation
+import Network
+
+@objc
+public class Downloader: NSObject {
+    private var allowCellular: Bool = false
+    private var refreshModels: Bool = false
+    private let monitor = NWPathMonitor()
+    private var path: NWPath?
+    private var downloadQueue: [URL:(Error?, String?) -> Void] = [:]
+    lazy var pathUpdateHandler = { (path: NWPath) in
+        self.path = path
+        if self.isNetworkAvailable() && !self.downloadQueue.isEmpty {
+            print("Downloader: executing from pathUpdateHandler")
+            for (url, complete) in self.downloadQueue {
+                self.downloadModel(url, complete)
+            }
+            // Reset queue
+            self.downloadQueue = [:]
+        }
+    }
+
+    init(allowCellular: Bool, refreshModels: Bool) {
```

Comment:
Since you're both defaulting the property values as well as requiring them in the constructor, you should just do one or the other. In this case, default args would be clearer to the caller.
```suggestion
    init(allowCellular: Bool = false, refreshModels: Bool = false) {
```